arcgis-ios-sdk-samples/Augmented reality/Display scenes in tabletop AR/DisplayScenesInTabletopAR.swift

```diff
@@ -0,0 +1,226 @@
+
+        // Set the origin camera
+        self.arView.originCamera = newCam
+    }
+}
+
+// MARK: - position the scene on touch
+extension DisplayScenesInTabletopAR: AGSGeoViewTouchDelegate {
+    public func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Only let the user place the scene once
+        if hasPlacedScene {
+            return
+        }
+        // Use a screen point to set the initial transformation on the view.
+        if self.arView.setInitialTransformation(using: screenPoint) {
+            configureSceneForAR()
+        } else {
+            presentAlert(message: "Failed to place scene, try again")
+        }
+    }
+
+    private func enableTapToPlace() {
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.helpLabel.isHidden = false
+            self.helpLabel.text = "Tap a surface to place the scene"
+
+            // Wait for the user to tap to place the scene
+            self.arView.sceneView.touchDelegate = self
+        }
+    }
+}
+
+// MARK: - tracking status display
+extension DisplayScenesInTabletopAR: ARSCNViewDelegate {
+    public func session(_ session: ARSession, cameraDidChangeTrackingState camera: ARCamera) {
+        switch camera.trackingState {
+        case .normal:
+            if hasPlacedScene {
+                helpLabel.isHidden = true
+            } else if !hasFoundPlane {
+                helpLabel.isHidden = false
+                helpLabel.text = "Keep moving your phone"
+            }
+        case .notAvailable:
+            helpLabel.text = "Location not available"
+        case .limited(let reason):
+            switch reason {
+            case .excessiveMotion:
+                helpLabel.text = "Try moving your phone more slowly"
+                helpLabel.isHidden = false
+            case .initializing:
+                helpLabel.text = "Keep moving your phone"
+                helpLabel.isHidden = false
+            case .insufficientFeatures:
+                helpLabel.text = "Try turning on more lights and moving around"
+                helpLabel.isHidden = false
+            case .relocalizing:
+                // this won't happen as this sample doesn't use relocalization
+                break
+            @unknown default:
+               break
+            }
+        }
+    }
+
+    // MARK: - Wait for plane before enabling scene
+    public func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
+        guard anchor as? ARPlaneAnchor != nil else { return }
+
+        // If we haven't placed a scene yet, enable tapping to place a scene and draw the ARKit plane found
+        if !hasPlacedScene {
+            hasFoundPlane = true
+            enableTapToPlace()
+            visualizePlane(renderer, didAdd: node, for: anchor)
+        }
+    }
+
+    // MARK: - Optional plane visualization
```

Comment:
I had this in originally, but I wonder if the 'optional' part might be confusing. Maybe just say 'MARK: - Plane visualization'?